rpc/ws/client.go

```diff
@@ -36,6 +37,8 @@ type result interface{}
 
 type Client struct {
 	rpcURL                  string
+	opt                     *Options
+	ctx                     context.Context
```

Comment:
context shouldn't be dependency of any struct. I.e, each action requires a context and should be determined as either background (run forever) or someone will cancel this action by using context. 

In this sense, each action (connect/ read/ write to ws) will required user's input context. As I can see context is only used for dial-up action. Hence, just use background for it.

Reply:
I hold `ctx` to use it in `dialer.DialContext()`.

Reply:
Imo using ctx in struct may cause confuse, take a look into this url https://go.dev/blog/context-and-structs